include/chemfiles/capi/shared_allocator.hpp

```diff
@@ -2,9 +2,7 @@
 #define CHFL_SHARED_ALLOCATOR_HPP
 
 #include <vector>
-#include <memory>
 #include <functional>
-#include <algorithm>  // IWYU pragma: keep
```

Comment:
I don't really know, this include was added the first time I ran include-what-you-use on the code, in db19ed8fbdaca209d1ca71b01a51a6386befe4e4. It might have been to workaround a bug with include-what-you-use, I remember it being pretty fiddly at the time.